File: MATLAB/pe060.m



clc; clear; close;
p = primes(10000);


for a = 1:length(p)
    A = p(a);

    for b = a:length(p)
        B = p(b);
        
        pairA1 = isprime(str2double([num2str(A) num2str(B)]));
        pairB1 = isprime(str2double([num2str(B) num2str(A)]));

        if(pairA1 == false || pairB1 == false); continue; end

        for c = b:length(p)
        C = p(c);

        pairA2 = isprime(str2double([num2str(A) num2str(C)]));
        pairB2 = isprime(str2double([num2str(B) num2str(C)]));

        pairC1 = isprime(str2double([num2str(C) num2str(A)]));
        pairC2 = isprime(str2double([num2str(C) num2str(B)]));


         if(pairA2 == false || pairB2 == false || ...
            pairC1 == false || pairC2 == false)
             continue
         end



            for d = c:length(p)
            D = p(d);
                
            
            pairA3 = isprime(str2double([num2str(A) num2str(D)]));
            pairB3 = isprime(str2double([num2str(B) num2str(D)]));
            pairC3 = isprime(str2double([num2str(C) num2str(D)]));
            
            pairD1 = isprime(str2double([num2str(D) num2str(A)]));
            pairD2 = isprime(str2double([num2str(D) num2str(B)]));
            pairD3 = isprime(str2double([num2str(D) num2str(C)]));


            if(pairA3 == false || pairB3 == false || pairC3 == false || ...
               pairD1 == false || pairD2 == false || pairD3 == false)
               continue 
            end

            
                for e = d:length(p)
                    E = p(e);
    
                    
                pairA4 = isprime(str2double([num2str(A) num2str(E)]));
                pairB4 = isprime(str2double([num2str(B) num2str(E)]));
                pairC4 = isprime(str2double([num2str(C) num2str(E)]));
                pairD4 = isprime(str2double([num2str(D) num2str(E)]));
                
                pairE1 = isprime(str2double([num2str(E) num2str(A)]));
                pairE2 = isprime(str2double([num2str(E) num2str(B)]));
                pairE3 = isprime(str2double([num2str(E) num2str(C)]));
                pairE4 = isprime(str2double([num2str(E) num2str(D)]));
    
    
                    if(pairA1 == true && pairA2 == true && pairA3 == true && pairA4 == true && ...
                       pairB1 == true && pairB2 == true && pairB3 == true && pairB4 == true && ...
                       pairC1 == true && pairC2 == true && pairC3 == true && pairC4 == true && ...
                       pairD1 == true && pairD2 == true && pairD3 == true && pairD4 == true && ...
                       pairE1 == true && pairE2 == true && pairE3 == true && pairE4 == true)
        
                        num1 = A
                        num2 = B
                        num3 = C
                        num4 = D
                        num5 = E
        
                        Sum = num1 + num2 + num3 + num4 + num5
        
        
                        return 
                    end
                    
                end
            end
        end
    end
end

File: C++/pe060.cpp
#include <iostream>
#include <vector>
#include <cmath>

bool isPrime(int n) {
    if (n <= 1) return false;
    if (n <= 3) return true;
    if (n % 2 == 0 || n % 3 == 0) return false;

    for (int i = 5; i * i <= n; i += 6) {
        if (n % i == 0 || n % (i + 2) == 0) return false;
    }
    return true;
}

bool isConcatPrime(int a, int b) {
    std::string s1 = std::to_string(a);
    std::string s2 = std::to_string(b);
    int num1 = std::stoi(s1 + s2);
    int num2 = std::stoi(s2 + s1);
    return isPrime(num1) && isPrime(num2);
}

int main() {
    const int limit = 10000;
    std::vector<int> primes;
    
    for (int i = 2; i <= limit; ++i) {
        if (isPrime(i)) {
            primes.push_back(i);
        }
    }

    for (size_t a = 0; a < primes.size(); ++a) {
        int A = primes[a];
        
        for (size_t b = a; b < primes.size(); ++b) {
            int B = primes[b];
            
            if (!isConcatPrime(A, B)) continue;

            for (size_t c = b; c < primes.size(); ++c) {
                int C = primes[c];
                
                if (isConcatPrime(A, C) && isConcatPrime(B, C)) {
                    for (size_t d = c; d < primes.size(); ++d) {
                        int D = primes[d];
                        
                        if (isConcatPrime(A, D) && isConcatPrime(B, D) && isConcatPrime(C, D)) {
                            for (size_t e = d; e < primes.size(); ++e) {
                                int E = primes[e];
                                
                                if (isConcatPrime(A, E) && isConcatPrime(B, E) &&
                                    isConcatPrime(C, E) && isConcatPrime(D, E)) {
                                    std::cout << "Found a set: " << A << " " << B << " " << C << " " << D << " " << E << std::endl;
                                    int Sum = A + B + C + D + E;
                                    std::cout << "Sum of the set: " << Sum << std::endl;
                                    return 0;
                                }
                            }
                        }
                    }
                }
            }
        }
    }
    
    return 0;
}
